### PR TITLE
Added the announcement message type

### PR DESCRIPTION
**Pull request by mmattbtw**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [x] Documentation in docs/ or install-docs/ was updated, if applicable
- [x] I have tested all changes

### CODE REVIEW

File: pajbot/bot.py [hunk 1]
```diff
@@ -785,9 +789,16 @@ def safe_say(self, message: str, channel: Optional[str] = None) -> None:
...
 
+    def safe_announcement(self, message: str, channel: Optional[str] = None) -> None:
+        if not self.is_bad_message(message):
+            self.announcement(message, channel)
+
     def me(self, message: str, channel: Optional[str] = None) -> None:
         self.say("/me " + message[: CHARACTER_LIMIT - 4], channel=channel)
 
+    def announcement(self, message: str, channel: Optional[str] = None) -> None:
+        self.say("/announce " + message[: CHARACTER_LIMIT - 4], channel=channel)
```
Comment (pajlada) [inline pajbot/bot.py:800]: The `-4` from `def me` is to account for the characters that we prefix (`/me `), so for `/announce` we'd need to cut 10 characters instead
```suggestion
        self.say("/announce " + message[: CHARACTER_LIMIT - 10], channel=channel)
```

**Review by pajlada — CHANGES_REQUESTED** [review]
Feature looks good to me, happy to add it. I think we should call it `announce` instead though, if that's something you can do just poke me when it's done and I'll review&merge this in. If not, let me know and I'll make those changes on your branch :+1: Thank you!

File: pajbot/utils/clean_up_message.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from typing import Optional
 
 # list of twitch commands permitted, without leading slash or dot
-permitted_commands = {"me"}
+permitted_commands = {"me", "announce"}
```
Comment (pajlada) [inline pajbot/utils/clean_up_message.py:4]: This list is for commands that we don't care if a user uses with user-input - changing a command from a normal privmsg to an action is fine, but using announce is too privileged

```suggestion
permitted_commands = {"me"}
```

File: templates/admin/create_command.html [hunk 3]
```diff
@@ -39,6 +39,7 @@ <h2>Create Command</h2>
             <select class="ui dropdown" name="reply">
                 <option value="say">Say</option>
                 <option value="me">Me</option>
+		<option value="announce">Announce</option>
```
Comment (pajlada) [inline templates/admin/create_command.html:42]: Use spaces instead of tabs

File: templates/admin/create_timer.html [hunk 4]
```diff
@@ -33,6 +33,7 @@ <h2>Create Timer</h2>
             <select class="ui dropdown" name="message_type">
                 <option value="say" {{ 'selected' if timer and timer.action.subtype == 'say' else '' }}>Say</option>
                 <option value="me" {{ 'selected' if timer and timer.action.subtype == 'me' else '' }}>Me</option>
+		<option value="announce" {{ 'selected' if timer and timer.action.subtype == 'announce' else '' }}>Announce</option>
```
Comment (pajlada) [inline templates/admin/create_timer.html:36]: Use spaces instead of tabs

**Review by pajlada — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,7 @@ Remember to bring your dependencies up to date with `./scripts/venvinstall.sh` w
 - Major: Remove pleblist API endpoints. (#1809)
 - Major: Remove pleblist pages. (#1809)
 - Minor: Set the ignore, admincommands, dbmanage & debug modules as hidden due to their un-toggleable and un-configurable nature. (#1835)
+- Minor: Added the `announce` message type. (#1847)
```
Comment (ALazyMeme) [inline CHANGELOG.md:13]: Recent changelog entries should be added on top of the categorized list.

```suggestion
- Minor: Added the `announce` message type. (#1847)
- Minor: Set the ignore, admincommands, dbmanage & debug modules as hidden due to their un-toggleable and un-configurable nature. (#1835)
```

File: pajbot/bot.py [hunk 6]
```diff
@@ -716,6 +716,8 @@ def send_message_to_user(
             self.whisper(user, message)
         elif method == "me":
             self.me(message)
+        elif method == "announce":
+            self.announce(message)
```
Comment (ALazyMeme) [inline pajbot/bot.py:720]: Announcements are announcements, they aren't to a specific user, but to the entire chat. I don't think this is appropriate in this function.

**Review by ALazyMeme — CHANGES_REQUESTED** [review]

File: pajbot/modules/quest.py [hunk 7]
```diff
@@ -24,15 +24,15 @@ class QuestModule(BaseModule):
             type="options",
             required=True,
             default="say",
-            options=["say", "whisper", "me", "reply"],
+            options=["say", "whisper", "me", "announce", "reply"],
```
Comment (ALazyMeme) [inline pajbot/modules/quest.py:27]: Wouldn't be a good idea for this command
```suggestion
            options=["say", "whisper", "me", "reply"],
```

File: pajbot/modules/quest.py [hunk 8]
```diff
@@ -24,15 +24,15 @@ class QuestModule(BaseModule):
...
         ),
         ModuleSetting(
             key="action_tokens",
             label="MessageAction for !tokens",
             type="options",
             required=True,
             default="whisper",
-            options=["say", "whisper", "me", "reply"],
+            options=["say", "whisper", "me", "announce", "reply"],
```
Comment (ALazyMeme) [inline pajbot/modules/quest.py:35]: Neither this command
```suggestion
            options=["say", "whisper", "me", "reply"],
```

**Review by ALazyMeme — CHANGES_REQUESTED** [review]

**Review by pajlada — APPROVED** [review]

**Review by ALazyMeme — APPROVED** [review]
yep

**Pull request merged**